cmd/mposter/main_test.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/mgurov/mposter/internal/assertions"
+	"github.com/mgurov/mposter/internal/testserver"
+)
+
+func TestSimpleRun(t *testing.T) {
+
+	result := execute(t, func(run *TestRun) {
+		run.input = "A\nB\nC"
+		run.path = "/path/"
+	})
+	result.AssertHttpAccessLog("POST /path/A\n" +
+		"POST /path/B\n" +
+		"POST /path/C\n")
+	result.AssertOutput("A OK\nB OK\nC OK\n")
+
+}
+
+func TestShouldSkipEmptyLines(t *testing.T) {
+
+	result := execute(t, func(run *TestRun) {
+		run.input = "\nA\n\nB\nC\n"
+	})
+	result.AssertHttpAccessLog("POST /A\nPOST /B\nPOST /C\n")
+	result.AssertOutput("A OK\nB OK\nC OK\n")
+}
+
+func TestShouldPathEncode(t *testing.T) {
+
+	result := execute(t, func(run *TestRun) {
+		run.input = "a#b+c"
+		run.path = "/path/"
+	})
+
+	result.AssertHttpAccessLog("POST /path/a%23b+c\n")
+
+	result.AssertOutput("a#b+c OK\n")
+}
+
+func TestShouldQueryEncode(t *testing.T) {
+
+	result := execute(t, func(run *TestRun) {
+		run.input = "a#b+c"
+		run.path = "/path?query="
+	})
+
+	result.AssertHttpAccessLog("POST /path?query=a%23b%2Bc\n")
+
+	result.AssertOutput("a#b+c OK\n")
+}
+
+func TestMultipleParametersSupport(t *testing.T) {
+	result := execute(t, func(run *TestRun) {
+		run.input = "A 1\nB 2\nC 3"
+		run.path = "/path/{{index . 0}}/sub/{{index . 1}}"
```

Comment:
Consider a more sane path template, e.g. `/path/{{0}}/sub/{{1}}` etc. with custom parsing for convenience.

Reply:
#6